### PR TITLE
[Api] Bug 101366 - Incorrect comment/discussion pagination

### PR DESCRIPTION
**Pull request by Matt-Beldon-CDS**
[*] Fixes comment/discussion returned total count

### CODE REVIEW

**Comment by Matt-Beldon-CDS** [comment]
Splits out getting total count and paginated records into two separate calls, trying to use dapper to query multiple rows and also map into an object seems impossible

**Review by JackWCDS — APPROVED** [review]

**Comment by Alex-Grubba-CDS** [comment]
Can we merge this PR? So we can test the fix. Quicker to test the fix and confirm/reject it than wait until Monday for review by Tom.

File: futurenhs.api/FutureNHS.Api/DataAccess/Database/Read/DiscussionsDataProvider.cs [hunk 1]
```diff
@@ -201,7 +214,7 @@ FROM Comment
                             });
 
             return GenerateDiscussionModelFromData(results).FirstOrDefault();
-        }
+        }
```
Comment (cds-tom-searle) [inline futurenhs.api/FutureNHS.Api/DataAccess/Database/Read/DiscussionsDataProvider.cs:217]: ws

Comment (Matt-Beldon-CDS) [reply futurenhs.api/FutureNHS.Api/DataAccess/Database/Read/DiscussionsDataProvider.cs:217]: Will address in next commit

**Pull request merged**